### PR TITLE
[DOC] GSoC Blog: Robin Roy (Community Bonding)

### PR DESCRIPTION
**Pull request by robinroy03**
My community bonding period blog.

Also fixed a 2021 blog typo. Codespell was flagging them.

### CODE REVIEW

File: docs/source/posts/2024/2024-05-28-week-0-robin.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+   :tags: google
+   :category: gsoc
+
+Hi, I'm Robin and I'm a 2nd year CS undergrad from Vellore Institute of Technology, Chennai. During GSoC '24 my work will be to build an LLM chatbot which will help the community by answering their questions.
+Scientific visualization is often complicated and hard for people to get used to - "Although 3D visualization technologies are advancing quickly, their sophistication and focus on non-scientific domains makes it hard for researchers to use
+them. In other words, most of the existing 3D visualization and computing APIs are low-level
+(close to the hardware) and made for professional specialist developers." [`FURY-Statement of Need <https://joss.theoj.org/papers/10.21105/joss.03384>`_]. FURY is our effort to bridge this gap with an easy-to-use API. With LLMs, the goal is to take this one step further and make it even simpler for people to get started. By reducing the barrier to entry, we can bring more people into this domain. Visualization should not be the most time-consuming thing for an engineer/researcher, it is supposed to just happen and help them accelerate faster.
+
+My Community Bonding Work
+--------------------------
```
Comment (itellaetxe) [inline docs/source/posts/2024/2024-05-28-week-0-robin.rst:15]: Remove one separator, separator line should be the same length as the title length

**Review by itellaetxe — COMMENTED** [review]
Nice work in your CB period Robin!! Super productive

The blog LGTM, only one small issue with the title markers

**Comment by deka27** [comment]
I think you should squash your commits.

**Comment by robinroy03** [comment]
@deka27 commits squashed :+1:

**Review by deka27 — APPROVED** [review]

**Review by WassCodeur — APPROVED** [review]

File: docs/source/posts/2024/2024-05-28-week-0-robin.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
+Week 0: Community Bonding!
+==========================
+
+.. post:: May 29 2024
+   :author: Robin Roy
+   :tags: google
+   :category: gsoc
+
+Hi, I'm Robin and I'm a 2nd year CS undergrad from Vellore Institute of Technology, Chennai. During GSoC '24 my work will be to build an LLM chatbot which will help the community by answering their questions.
+Scientific visualization is often complicated and hard for people to get used to - "Although 3D visualization technologies are advancing quickly, their sophistication and focus on non-scientific domains makes it hard for researchers to use
```
Comment (skoudoro) [inline docs/source/posts/2024/2024-05-28-week-0-robin.rst:10]: please, add an empty line, it is dense. 

image

File: docs/source/posts/2024/2024-05-28-week-0-robin.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+them. In other words, most of the existing 3D visualization and computing APIs are low-level
+(close to the hardware) and made for professional specialist developers." [`FURY-Statement of Need <https://joss.theoj.org/papers/10.21105/joss.03384>`_]. FURY is our effort to bridge this gap with an easy-to-use API. With LLMs, the goal is to take this one step further and make it even simpler for people to get started. By reducing the barrier to entry, we can bring more people into this domain. Visualization should not be the most time-consuming thing for an engineer/researcher, it is supposed to just happen and help them accelerate faster.
```
Comment (skoudoro) [inline docs/source/posts/2024/2024-05-28-week-0-robin.rst:12]: please add empty line

File: docs/source/posts/2024/2024-05-28-week-0-robin.rst [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+It works. But Google Colab won't be 24x7 active (it will close after some time when the tab is closed) and it was very hacky.
+
+   .. raw:: html
+
+      <iframe width="640" height="390" src="https://drive.google.com/file/d/1qNLtXxAMlLQ8xO8jfV0keRtskvcsj-fC/preview" frameborder="0" allowfullscreen></iframe>
+
+2) **Ollama on Kaggle**
+
+Same as above, same issues. Talked with my mentor Mohamed and skipped implementation.
```
Comment (skoudoro) [inline docs/source/posts/2024/2024-05-28-week-0-robin.rst:36]: add github link for mohamed

File: docs/source/posts/2024/2024-05-28-week-0-robin.rst [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+3) **GGUF models with ctransformers on HuggingFace**
```
Comment (skoudoro) [inline docs/source/posts/2024/2024-05-28-week-0-robin.rst:38]: GGUF ? please be explicit at least once.

File: docs/source/posts/2024/2024-05-28-week-0-robin.rst [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+The way it works is by taking a gguf model and then inferencing using the ctransformers library from HF. An endpoint will be exposed using flask/fastapi.
+
+It had issues like not all models were working, and ctransformers did not support all models. And the models that do work were slow on my machine. Local testing was a nightmare and inference speed on HF was also very slow.
+
+4) **GGUF models with llama-cpp-python, hosted on HF**
+
+I used langchain wrapper over ollama-cpp-python to inference GGUF models. This one was able to handle all GGUF models, and local testing was okayish. When I tried handling concurrent requests, it crashed and gave segfs. I fixed segfs later by increasing gunicorn workers (Gunicorn was the WSGI server I used).
```
Comment (skoudoro) [inline docs/source/posts/2024/2024-05-28-week-0-robin.rst:46]: segfs? What does it mean?

**Review by skoudoro — COMMENTED** [review]
Hi @robinroy03,

Thank for this! See below my first comment

File: docs/source/posts/2024/2024-05-28-week-0-robin.rst [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+TLDR: This one worked!
+
+   .. raw:: html
+
+      <iframe width="640" height="390" src="https://drive.google.com/file/d/17yxdw169uqLlw6WKfi--bWEUQArJk7i2/preview" frameborder="0" allowfullscreen></iframe>
+
+Ollama was perfect, it works like a charm on my machine and the ecosystem is also amazing (the people on their discord server are super kind). I knew I had to try ollama on HF.
+I was unable to initially run ollama and provide an endpoint. My dockerfile builds were all failing. Later mentor Serge told me to use the official Ollama image (till then I was using the Ubuntu base image).
+
+I managed to get the dockerfile running locally, but still, the HF build was failing. Then I took help from HF community. They told me it was HF blocking some ports, so to try different ports. This is when I came across another ollama server repo, and it was using Ubuntu as the base image. I studied that code and modified my dockerfile. It was adding an env variable to repo settings that I missed. My current dockerfile is just 5 lines and it works well.
```
Comment (skoudoro) [inline docs/source/posts/2024/2024-05-28-week-0-robin.rst:64]: That's great! good job for asking for help, experimenting and finding solution.  👍

**Review by skoudoro — COMMENTED** [review]
Hi @robinroy03,

See below my last comment. 

Can you also rebase your PR. Thank you

**Comment by m-agour** [comment]
Hi @skoudoro @robinroy03 
This blog post It renders without any problems and videos are working great!

**Review by skoudoro — APPROVED** [review]
All ok, merging

**Pull request merged**